### ISSUE TITLE
X11 not working for some 32-bit powermac users

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#385]
Reported on [the list](https://lore.kernel.org/linuxppc-dev/d5d0b396-7408-bdae-cf50-4b5f4e7b3184@csgroup.eu/).

Commit torvalds/linux@d3ccc9781560 identified as the cause.
May also require `CONFIG_USER_NS=y` and particular package versions to reproduce.

**Comment by chleroy** [comment]
```
static inline int __get_user_sigset(sigset_t *dst, const sigset_t __user *src)
{
	BUILD_BUG_ON(sizeof(sigset_t) != sizeof(u64));

	return __get_user(dst->sig[0], (u64 __user *)&src->sig[0]);
}
```

`dst->sig[0]` is 32 bits on ppc32, so it won't work.

Should work by replacing it to `*(u64*)&dst->sig[0]`

**Comment by mpe** [comment]
Fixed in v5.16-rc2 torvalds/linux@5499802b2284.
